woocommerce/woocommerce-ios
# Shipping Labels: Refactor package details view model with Combine

Closes #4666 

# Description
Currently the view model of Package Details has duplicated calls to `setTotalWeight` and `setItemRows`. Since we're already using publishers, I figure this is a nice opportunity to make use of Combine. This PR aims to refactor these functions by observing changes in relevant variables and re-calculate data when appropriate.

# Changes
## Configuring item rows
- Updated `generateItemsRows` method to accept products and product variations as input instead of reading from class properties directly.  This is to make sure of the single source of truth for the method.
- Create new method `configureItemsRows` to observe changes in products and variations and get results from `generateItemsRows` to assign to `itemsRows` directly. This method is called directly from the initializer of the class.
- Remove duplicated calls to `generateItemsRows`.

## Configuring total weight
- Renamed `setTotalWeight` to `calculateTotalWeight`, and updated it to accept products, product variations and an optional custom package as input. The result weight will be calculated solely on these input.
- Added an optional variable `initialTotalWeight` as a new private property for the view model.
- Added a new method `configureTotalWeight` to observe changes in selected custom package, products and variations to calculate total weight accordingly. This observing is only handled when the package weight is not manually input, and there's also a check for initial total weight to update `isPackageWeightEdited` appropriately.
- Updated unit tests for `isPackageDetailsDoneButtonEnabled` to reflects the function correctly.

# Testing
Unit tests should still pass after the refactoring. When running the feature, everything should still work as expected:
1. Navigate to Orders tab.
2. Select an order with status Processing.
3. Select Create Shipping Label.
4. Configure package details: notice that order items are correct, and package weight is updated correctly when switching between packages.

Update release notes:
